### DISCUSSION TITLE
Disable `brew cleanup` on specific formulae

### DISCUSSION THREAD

**Discussion by voldemortensen** [Homebrew/discussions#627] [Everyday usage]
I have a few formulae that I like holding on to a few different versions because I need to switch between them fairly often. I know I can disable cleanup entirely with `HOMEBREW_NO_INSTALL_CLEANUP=1`, but that ends up eating disk space if I don't handle the cleanup myself.

Is there are way to disable clean for specific formulae so most everything will still be cleaned?

**Comment by SMillerDev** [comment]
This sounds more like a use case for `brew extract`

**Reply by voldemortensen** [reply]
I know it can be done that way, it's just a lot more tedious to maintain a separate tap with formula for different versions. That also removes the ability to `brew switch <formula> <version>`. It would become `brew unlink <formula>@<version>` and then `brew link <formula>@<version>`.

**Reply by gromgit** [reply]
`brew switch` was disabled some time back, so you'll have to use separate formulae.
